languages/julia/julia-tutorial/source-notebooks/Julia-Parallel-Computing.ipynb

```diff
@@ -0,0 +1,1977 @@
+     ]
+    }
+   ],
+   "source": [
+    "@btime run_mci_dist2();"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ef0e67c6",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "subslide"
+    }
+   },
+   "source": [
+    "## `@spawnat` Macro\n",
+    "\n",
+    "**WARNING:** The entire expression is sent to the worker process before anything in the expression is executed. This can cause performance issues if we need a small part of a big object or array."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "id": "8f28213b",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "@everywhere struct MyData\n",
+    "    Data::Vector{Float64}\n",
+    "    N::Int\n",
+    "end\n",
+    "function slow(my_data::MyData)\n",
+    "    return fetch(@spawnat(2, mean(rand(my_data.N))))\n",
+    "end;"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "id": "155705f5",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "  1.740 ms (100 allocations: 4.13 KiB)\n"
+     ]
+    }
+   ],
+   "source": [
+    "large_data = MyData(rand(1000000), 5)\n",
+    "@btime slow(large_data);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8f27704f",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "subslide"
+    }
+   },
+   "source": [
+    "## `@spawnat` Macro\n",
+    "\n",
+    "**WARNING:** The entire expression is sent to the worker process before anything in the expression is executed. This can cause performance issues if we need a small part of a big object or array.\n",
+    "\n",
+    "This is easily fixed using a local variable:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "id": "b5df1b86",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "function fast(my_data::MyData)\n",
+    "    n = my_data.N\n",
+    "    return fetch(@spawnat(2, mean(rand(n))))\n",
+    "end;"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "id": "9218427f",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "  164.427 μs (92 allocations: 3.87 KiB)\n"
+     ]
+    }
+   ],
+   "source": [
+    "@btime fast(large_data);"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "id": "b6ee9908",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "skip"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "large_data = nothing"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "48fbd1d5",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "skip"
+    }
+   },
+   "source": [
+    "**TODO:** Do I want to do this bit?"
```

Comment:
Do we need this TODO?

Reply:
So that section of slides in the notebook didn't make it into the markdown file. Hence the TODO. I changed it to something a bit more meaningful--namely a note that says the slides on SharedArrays could be covered in a future version.